contributed_definitions/nyaml/NXpiezo_config_spm.yaml

```diff
@@ -0,0 +1,59 @@
+category: base
+doc: |
+  A base class to describe piezo settings for scanning probe microscopy.
+
+  The material of the Piezoelectric scanner is composed of polycrystalline solids and 
+  attached at the head of the tip. The material is highly sensitive to applied voltage. 
+# No need to create this ´base class rather we can define in spm application definition.
+NXpiezo_config_spm(NXobject):
+  piezo_material(NXpiezoelectric_material):
+    doc: |
+      The material description and properties of the piezoelectric scanner materials.
+  active_calib:
+    doc: |
+      The name of the calibration type (e.g. LHe).
+  calib_N(NX_NUMBER):
+    exists: optional
+    doc: |
+      N denotes X or Y or Z. There are three directions X, Y, and Z for calibration, along with
+      three available controls: Calibration (m/V), Range (m), and HV gain each. Only
+      two of these parameters are required to define the calibration. Consequently, when any
+      value is changed, one of the other values will be automatically updated. (e.g. calib_X = 3.8E-9)
+  hv_gain_N(NX_NUMBER):
+    exists: optional
+    doc: |
+      N denotes X or Y or Z. In some systems, there is an HV gain readout feature. For
+      these systems, the HV gain should be automatically adjusted whenever the gain is
+      changed at the high voltage amplifier (e.g. 14.5).
+    # (e.g. 0.318343)
+  tilt_N(NX_NUMBER):
+    exists: optional
+    unit: NX_ANGLE
+    doc: |
+      N denotes X or Y. There are 2 parameters in X and Y directions, and tilt needs to be
+      adjusted according to the actual surface. (in degrees, first order correction).
+  
+  curvature_radius_N(NX_NUMBER):
+    exists: optional
+    unit: NX_LENGTH
+    doc: |
+      N denotes X or Y. There are 2 parameters in X and Y directions. It can be set
+      approximately to the length of the piezo tube.
+  2nd_order_corr_N(NX_NUMBER):
+    exists: optional
+    doc: |
+      N denotes X or Y. There are 2 parameters in X and Y directions. If you know them,
+      you can enter the 2nd order piezo characteristics to compensate for it. The
+      following equation shows the interpretation of the 2nd order correction parameter:
+      For the X-piezo: "Ux = 1/cx · X + cxx · X2" with units:
+      [V] = [V/m] · [m] + [V/m2] · [m2] where cx is the calibration of the piezo X and cxx
+      is the 2nd order correction. (V/m^2) (e.g. 0E+0).
+  drift_N(NX_NUMBER):
```

Comment:
It is is a speed, shouldn't it have unit `NX_VELOCITY`?

Reply:
Noted!